src/test/java/api/RelatedInstanceTest.java

```diff
@@ -0,0 +1,133 @@
+package api;
+
+import static api.support.InstanceSamples.nod;
+import static io.vertx.core.json.JsonObject.mapFrom;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.folio.inventory.domain.instances.Instance;
+import org.folio.inventory.domain.instances.RelatedInstance;
+import org.junit.Test;
+
+import api.support.ApiTests;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class RelatedInstanceTest extends ApiTests {
+
+  @Test
+  public void testRelatedInstance() {
+    String id = UUID.randomUUID().toString();
+    String instanceId = UUID.randomUUID().toString();
+    String relatedInstanceId = UUID.randomUUID().toString();
+    String relatedInstanceTypeId = UUID.randomUUID().toString();
+    RelatedInstance relatedInstance = new RelatedInstance(
+      id,
+      instanceId,
+      relatedInstanceId,
+      relatedInstanceTypeId
+    );
+
+    JsonObject relatedInstanceJson = mapFrom(relatedInstance);
+
+    RelatedInstance sameRelatedInstance = RelatedInstance.from(relatedInstanceJson, instanceId);
+
+    assertEquals(id, sameRelatedInstance.id);
+    assertEquals(instanceId, sameRelatedInstance.instanceId);
+    assertEquals(relatedInstanceId, sameRelatedInstance.relatedInstanceId);
+    assertEquals(relatedInstanceTypeId, sameRelatedInstance.relatedInstanceTypeId);
+
+    assertEquals(relatedInstance, sameRelatedInstance);
+
+    RelatedInstance inverseRelatedInstance = RelatedInstance.from(relatedInstanceJson, relatedInstanceId);
+
+    assertEquals(id, inverseRelatedInstance.id);
+    assertEquals(relatedInstanceId, inverseRelatedInstance.instanceId);
+    assertEquals(instanceId, inverseRelatedInstance.relatedInstanceId);
+    assertEquals(relatedInstanceTypeId, inverseRelatedInstance.relatedInstanceTypeId);
+
+    assertNotEquals(relatedInstance, inverseRelatedInstance);
+  }
+
+  @Test
+  public void canCreateInstanceWithRelatedInstances() throws Exception {
+
+    var instanceId = UUID.randomUUID();
+    var relatedInstanceId = UUID.randomUUID();
+
+    String relatedInstanceTypeId = relatedInstanceTypeFixture.related().getId();
+
+    var firstNodJson = nod(instanceId)
+        .put(Instance.TITLE_KEY, "An instance");
+
+    instancesClient.create(firstNodJson);
+
+    var thirdNodJson = nod(relatedInstanceId)
+        .put(Instance.TITLE_KEY, "A related instance")
+        .put(Instance.RELATED_INSTANCES_KEY, new JsonArray()
+            .add(createRelatedInstance(UUID.randomUUID(), relatedInstanceId, instanceId, relatedInstanceTypeId)));
```

Comment:
This sends an `id` property which currently isn't defined in the API documentation

Reply:
Yes, there is a bit of mismatch between the Instance schema for relatedInstances and what is actual. Will require multiple representations of related instance; one that is placed in relatedInstances on the Instance and another for the entity retrieved from mod-inventory-storage related-instances endpoint.

I am concerned of this difference and not sure what is the best approach to handle. Keep in mind there is already a slight mutation of the related instances placed on the instance response in which the instanceId and relatedInstanceId are swapped is not matching the instance id in the expected "directionality".

Reply:
> Yes, there is a bit of mismatch between the Instance schema for relatedInstances and what is actual. 

What do you mean by actual? Is it the storage representation of related instances?

> Will require multiple representations of related instance; one that is placed in relatedInstances on the Instance and another for the entity retrieved from mod-inventory-storage related-instances endpoint.

Yes, I think that is inherent in trying to hide the artefacts of 

> I am concerned of this difference and not sure what is the best approach to handle.

What concerns do you have about the difference?

I think there needs to be a mapping. I would expect there is a mapping in instance relationships. Am I wrong about that? Maybe that is why the `id` is exposed and the records are split between parent and child.

> Keep in mind there is already a slight mutation of the related instances placed on the instance response in which the instanceId and relatedInstanceId are swapped is not matching the instance id in the expected "directionality".

I understand that. That is the nature of the domain model and the way the storage / business logic split is achieved.

I don't know of a way to avoid it.